cppguide.html

```diff
@@ -1719,6 +1719,9 @@ <h3 id="Inputs_and_Outputs">Inputs and Outputs</h3>
 improve readability, and often provide the same or better
 performance.</p>
 
+<p>Prefer to return by value or, failing that, return by reference.
+Avoid returning a pointer unless it can be null.</p>
```

Comment:
```suggestion
Avoid returning a pointer unless you take care of the case if it is null.</p>
```
otherwise it could be misunderstood.